app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala

```diff
@@ -69,4 +70,39 @@ class RecordsRepository @Inject() (
       )
       .toFuture()
       .map(_.getDeletedCount > 0)
+
+  def saveRecord(record: GoodsItemRecords): Future[Boolean] =
+    collection
+      .replaceOne(
+        filter = byRecordId(record.recordId),
+        replacement = record,
+        options = ReplaceOptions().upsert(true)
+      )
+      .toFuture()
+      .map(_ => true)
+
+  def update(eori: String, recordId: String, updateRequest: UpdateRecordRequest): Future[Option[GoodsItemRecords]] =
+    get(recordId).flatMap {
+      case Some(existingRecord) =>
+        val updatedRecord = existingRecord.copy(
+          traderRef = updateRequest.traderRef.getOrElse(existingRecord.traderRef),
+          comcode = updateRequest.comcode.getOrElse(existingRecord.comcode),
+          goodsDescription = updateRequest.goodsDescription.getOrElse(existingRecord.goodsDescription),
+          countryOfOrigin = updateRequest.countryOfOrigin.getOrElse(existingRecord.countryOfOrigin),
+          category = updateRequest.category.getOrElse(existingRecord.category),
+          assessments = updateRequest.assessments.orElse(existingRecord.assessments),
+          supplementaryUnit = updateRequest.supplementaryUnit.orElse(existingRecord.supplementaryUnit),
+          measurementUnit = updateRequest.measurementUnit.orElse(existingRecord.measurementUnit),
+          comcodeEffectiveFromDate =
+            updateRequest.comcodeEffectiveFromDate.getOrElse(existingRecord.comcodeEffectiveFromDate),
+          comcodeEffectiveToDate = updateRequest.comcodeEffectiveToDate.orElse(existingRecord.comcodeEffectiveToDate)
+        )
+
+        saveRecord(updatedRecord).map {
+          case true  => Some(updatedRecord)
+          case false => None
```

Comment:
saveRecord returns `.map(_ => true)` 
Therefore `case false => None` is unreachable here